rmf_fleet_adapter/src/full_control/estimation.cpp

```diff
@@ -214,6 +214,12 @@ void estimate_state(
     TravelInfo& info)
 {
   std::string last_known_map = l.level_name;
+  if(!last_known_map.empty())
+  {
+    info.updater->update_position(last_known_map, {l.x, l.y, l.yaw});
```

Comment:
I'm concerned that this will effectively erase information about the last known waypoint, and it's not clear to me if that's a good thing or a bad thing.

Are there specific problematic scenarios that this has managed to help with? If so, maybe the implementation of this entire function needs to be reconsidered.

Reply:
Hi @mxgrey, I figured the new changes here and below would improve estimations for situations where the robot transitions from a waypoint on one level to another. I made these changes initially while debugging the issue with robots ignoring their initial lane conditions at  times but that problem was solved by a different fix. If these changes are not useful, I can close this PR.

Reply:
I don't want to jump to any conclusions about whether they're useful or not.

But if these changes  **are** substantive improvements, then I think that suggests many of the assumptions that I made in this implementation were bad, and the whole implementation should be reconsidered.

I'm very open to that possibility; I just think it would be good to have some experimental validation of that before diving into that rabbit hole.

Reply:
That's a fair point @mxgrey. I'll work on experimental validation of this before pursuing any further.